src/pykoala/corrections/sky.py

```diff
@@ -71,443 +187,640 @@ def pol_continuum(data, wavelength, pol_order=3, **polfit_kwargs):
             skymodel_var = self.intensity[:, np.newaxis, np.newaxis]
         elif data.ndim == 2 and self.intensity.ndim == 1:
             skymodel_intensity = self.intensity[np.newaxis, :]
             skymodel_var = self.intensity[np.newaxis, :]
+        elif data.ndim == 2 and self.intensity.ndim == 2:  
+            skymodel_intensity = self.intensity       
+            skymodel_var = self.variance
         else:
-            self.vprint(f"Data dimensions ({data.shape}) cannot be reconciled with sky mode ({self.intensity.shape})")
+            self.vprint(
+                f"Data dimensions ({data.shape}) cannot be reconciled with "
+                + f"sky mode ({self.intensity.shape})")
         data_subs = data - skymodel_intensity
         var_subs = variance + skymodel_var
         return data_subs, var_subs
 
     def substract_pca():
-        # TODO: Implement PCA substraction method
+        """
+        Placeholder for PCA subtraction method.
+
+        This method is not yet implemented.
+        """
         pass
     
     def vprint(self, *messages):
-        """Print a message"""
+        """
+        Print messages if `verbose` is True.
+
+        Parameters
+        ----------
+        *messages : str
+            Messages to be printed.
+        """
         if self.verbose:
             print("[SkyModel] ", *messages)
 
 
 
 class SkyOffset(SkyModel):
     """
-    Sky model based on a single RSS offset sky exposure
+    Sky model based on a single RSS offset sky exposure.
 
-    Description
-    -----------
     This class builds a sky emission model from individual sky exposures.
 
     Attributes
     ----------
-    - dc:
-        Data container used to estimate the sky
-    - exptime:
-        Data container net exposure time
+    dc : DataContainer
+        Data container used to estimate the sky.
+    exptime : float
+        Net exposure time from the data container.
     """
+
     def __init__(self, dc):
         """
+        Initialize the SkyOffset model with a data container.
 
         Parameters
         ----------
-        rss: RSS
-            Raw Stacked Spectra corresponding to the offset-sky exposure.
+        dc : DataContainer
+            Raw Stacked Spectra (RSS) corresponding to the offset-sky exposure.
         """
         self.dc = dc
         self.exptime = dc.info['exptime']
         super().__init__()
 
     def estimate_sky(self):
+        """
+        Estimate the sky emission model.
+
+        This method calculates the intensity and variance of the sky model using
+        percentiles, then normalizes them by the exposure time.
+        """
         self.intensity, self.variance = BackgroundEstimator.percentile(
             self.dc.intensity, percentiles=[16, 50, 84])
         self.intensity, self.variance = (
             self.intensity / self.exptime,
             self.variance / self.exptime)
 
-
 class SkyFromObject(SkyModel):
```

Comment:
Now that I think of it, wouldn't it make more sense to have an `RSS.estimate_sky` method that returns a `SkyModel`, or a `SkyModel.estimate_from_rss` method, or even a variant of the `SkyModel` constructor that accepts an `RSS` as an argument?

Reply:
Here's where our opinions diverge. I don't believe this should be a method of RSS because:
- I think a Cube can also be used for that purpose
- If we include this correction, why do we not include the rest? This would lead to a long list of methods within RSS and Cube...
I am not completely against your approach, but I need a very good reason for doing such a big refactoring.

Reply:
> I think a Cube can also be used for that purpose
Then the right place would be `SpectraContainer`.

> If we include this correction, why do we not include the rest?
Indeed.

> This would lead to a long list of methods within RSS and Cube
That doesn't particularly bother me, but now, thinking about it, many of these methods share a common pattern. In particular, I do like that all corrections fulfil `CorrectionBase`. A lot. I don't know... in the end, you may eventually make me change my mind. In any case, we certainly agree with no refactoring anything unless we are firmly convinced that it's worth the effort.